### PR TITLE
CSF 3 Well & Switch

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

The goal is to outline a few simple patterns for some easy components.
I've also taken some liberties in removing stories I think are covered well enough by Chromatic.
And I've tried to combine some of our render functions to instead utilize CSF 3's property spreading as they recommend in their docs.

See docs for v2 vs v3 comparisons: https://storybook.js.org/blog/component-story-format-3-0/

Open questions/opinions needed:
Do we like this style over the Template.bind({}) syntax? Do we like spreading all the props or would we prefer more stories with custom render functions instead. For instance, should our Default Switch story render children, or should it take children as args?
In both cases, you could override the children, however, it's a little different.
```
export const Default: ComponentStoryObj<typeof Switch> = {
  render: (args) => (
    <Switch {...args} />
  ),
  args: {children: <>Switch Label</>}
};
// two options here, omit the Default.args
export const NoLabel: ComponentStoryObj<typeof Switch> = {
  ...Default,
  args: {'aria-label': 'This switch has no visible label'},
  name: 'no label',
  parameters: {description: {data: 'Try me with a screen reader.'}}
};
// or include them
export const NoLabel: ComponentStoryObj<typeof Switch> = {
  ...Default,
  args: {...Default.args, children: null, 'aria-label': 'This switch has no visible label'},
  name: 'no label',
  parameters: {description: {data: 'Try me with a screen reader.'}}
};
```
vs
```
export const Default: ComponentStoryObj<typeof Switch> = {
  render: (args) => (
    <Switch {...args}>Switch Label</Switch>
  ),
  args: {}
};
// two options here now to override the children for our story with no labels
export const NoLabel: ComponentStoryObj<typeof Switch> = {
  ...Default,
  args: {children: null, 'aria-label': 'This switch has no visible label'},
  name: 'no label',
  parameters: {description: {data: 'Try me with a screen reader.'}}
};
// or
export const NoLabel: ComponentStoryObj<typeof Switch> = {
  render: (args) => <Switch {...args} />,
  args: {'aria-label': 'This switch has no visible label'},
  name: 'no label',
  parameters: {description: {data: 'Try me with a screen reader.'}}
};
```

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/switch/stories/Switch.stories.tsx [hunk 1]
```diff
@@ -10,91 +10,104 @@
  * governing permissions and limitations under the License.
  */
 
-import {action} from '@storybook/addon-actions';
-import React from 'react';
-import {storiesOf} from '@storybook/react';
+import {chain} from '@react-aria/utils';
+import {ComponentMeta, ComponentStoryObj} from '@storybook/react';
```
Comment (snowystinger) [inline packages/@react-spectrum/switch/stories/Switch.stories.tsx:14]: these only exist in the newer versions of storybook, that's why I updated them

File: packages/@react-spectrum/switch/stories/Switch.stories.tsx [hunk 2]
```diff
@@ -10,91 +10,104 @@
...
-  )
-  .add(
-    'isDisabled: true',
-    () => render({isDisabled: true})
-  )
-  .add(
-    'isEmphasized: true',
-    () => render({isEmphasized: true})
-  )
-  .add(
```
Comment (snowystinger) [inline packages/@react-spectrum/switch/stories/Switch.stories.tsx:44]: i have removed this story entirely, it's covered in chromatic and it wouldn't have behavioral differences from the regular disabled story

File: packages/@react-spectrum/switch/stories/Switch.stories.tsx [hunk 3]
```diff
@@ -10,91 +10,104 @@
...
+};
+
+export const NoLabel: ComponentStoryObj<typeof Switch> = {
+  ...Default,
+  args: {'aria-label': 'This switch has no visible label'},
+  name: 'no label',
+  parameters: {description: {data: 'Try me with a screen reader.'}}
+};
+
+export const ControlledImplementation: ComponentStoryObj<typeof Switch> = {
```
Comment (snowystinger) [inline packages/@react-spectrum/switch/stories/Switch.stories.tsx:105]: I noticed we were missing a story for controlled switches

File: packages/@react-spectrum/switch/stories/Switch.stories.tsx [hunk 4]
```diff
@@ -10,91 +10,104 @@
...
-function render(props = {}) {
-  return (
-    <Switch
-      onChange={action('change')}
-      {...props}>
-      Switch Label
-    </Switch>
-  );
-}
+export const Default: ComponentStoryObj<typeof Switch> = {
```
Comment (reidbarber) [inline packages/@react-spectrum/switch/stories/Switch.stories.tsx:34]: We might want to do something like this in every file:

`type SwitchStory = ComponentStoryObj<typeof Switch>;`

File: packages/@react-spectrum/switch/stories/Switch.stories.tsx [hunk 5]
```diff
@@ -10,91 +10,104 @@
...
+  render: (args) => (
+    <Switch {...args} />
+  ),
```
Comment (reidbarber) [inline packages/@react-spectrum/switch/stories/Switch.stories.tsx:37]: Don't think the explicit render is needed. Or is it for some reason? can't remember what we decided on

Comment (snowystinger) [reply packages/@react-spectrum/switch/stories/Switch.stories.tsx:37]: that's a good point, i guess, technically we do not need it

**Comment by LFDanLu** [comment]
I think I like the first example better, but it feels like a pretty small difference overall. Think I would have to try using this format for a more complex component that has a more involved setup/varied setups (aka the collection ones) to form a strong opinion. Definitely like this over the bind syntax though, that syntax was always a bit confusing to me.

**Review by reidbarber — APPROVED** [review]
LGTM, definitely a huge improvement.

**Review by LFDanLu — APPROVED** [review]
LGTM

**Pull request merged**